lib/src/rules/unnecessary_parenthesis.dart

```diff
@@ -84,6 +85,14 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   _Visitor(this.rule);
 
+  @override
+  void visitParenthesizedPattern(ParenthesizedPattern node) {
+    var parent = node.parent;
+    if (parent is DartPattern && parent.precedence < node.precedence) {
```

Comment:
This looks good.

We should explore whether we want to flag any use of a parenthesized pattern where the parent isn't another pattern. Seems like that would be a good case, but I'm not familiar enough with the patterns feature to know for sure.

Reply:
Nor am I.  I'll do a little digging.  Thanks!